forem/forem
# Ensuring the same policies for analytics

## What type of PR is this? (check all applicable)


## Description

Prior to this commit the following situation existed:

> The path /dashboard/analytics/org/:id requires user
> authentication (e.g. signed in). However, it does not enforce
> authorization. Anyone can see this page. The page, however, uses
> javascript to populate the data. So no information, aside from the org
> name associated with the :id leaks out. The javascript API end point
> enforces organization membership.
>
> I would expect that the authorization in the HTML rendering would be
> the same as the javascript API end point.

This commit ensures that the dashboards#analytics end point uses the
same policy logic as the API analytics end points.


## Related Tickets & Documents

Closes forem/forem/#16985

## QA Instructions, Screenshots, Recordings

If you are interested you're going to need to make quite a few users:

0. An organization (eg. "an org")
1. A user who is a member of an org
2. A user who is an admin of an org
3. A user who is an admin of another org
4. A super-user who is not an admin of any org

Then login as each of those and go to the organization's `dashboard/analytics/org/:id` (where :id is the database ID of the organization).  Only the admin of the org should see a page.

### UI accessibility concerns?

None.

## Added/updated tests?


## [Forem core team only] How will this change be communicated?
